src/Spago/Lock/Migrations/FromSnakeCase.purs

```diff
@@ -0,0 +1,106 @@
+module Spago.Lock.Migrations.FromSnakeCase where
+
+import Spago.Core.Prelude
+
+import Data.Array.NonEmpty as NonEmptyArray
+import Data.List (List(..))
+import Data.List as List
+import Data.String as String
+import Data.String.Regex as Regex
+import Data.String.Regex.Flags as RegexFlags
+import Data.String.Regex.Unsafe as UnsafeRegex
+import Data.Symbol (class IsSymbol, reflectSymbol)
+import Data.Traversable (sequence)
+import Prim.RowList as RL
+import Spago.Core.Config (Dependencies, ExtraPackage, RemotePackage, SetAddress)
+import Spago.Lock (LockEntry, Lockfile)
+import Spago.Yaml (YamlMigrationStep(..))
+import Type.Proxy (Proxy(..))
+
+migrateLockFromSnakeCase :: List YamlMigrationStep
+migrateLockFromSnakeCase = migrateLockFromSnakeCase' (Proxy :: Proxy Lockfile)
+
+class MigrateLockFromSnakeCase' :: Type -> Constraint
+class MigrateLockFromSnakeCase' a where
+  migrateLockFromSnakeCase' :: Proxy a -> List YamlMigrationStep
+
+-- SetAddress doesn't have any memebers using snake case
+instance MigrateLockFromSnakeCase' SetAddress where
+  migrateLockFromSnakeCase' _ = Nil
+
+instance MigrateLockFromSnakeCase' ExtraPackage where
+  migrateLockFromSnakeCase' _ = Nil
+
+-- Dependencies should not be modified
+instance MigrateLockFromSnakeCase' Dependencies where
+  migrateLockFromSnakeCase' _ = Nil
+
+instance MigrateLockFromSnakeCase' LockEntry where
+  migrateLockFromSnakeCase' _ = Nil
+
+-- Values under this field should not be modified
+instance 
+  ( MigrateLockFromSnakeCase' v
+  ) => MigrateLockFromSnakeCase' (Map k v) where
+  migrateLockFromSnakeCase' _ = List.singleton $
+    YamlMigrationStep
+      { key: "foo"
+      , toPrevKey: Nothing
+      , updateValue: migrateLockFromSnakeCase' (Proxy :: Proxy v)
+      }
```

Comment:
Third, with the current design, this instance is incorrect as made obvious via `.workspace.packages.<package>.test_dependencies`. Similar to what I do with the sequence values, I'd need to modify `YamlMigrationStep` to specify whether the next value has a known key or whether one should just iterate through all of its keys.